src/App.js

```diff
@@ -26,27 +46,31 @@ function App() {
 
   history.navigate = useNavigate();
   history.location = useLocation();
-  const theme=useMemo(()=>{
-    if(store?.theme === 'dark') 
-    return {
-      background:'#1b1d21',
-      color:'white'
-    }
-    return {background:'#f3f6f9'}
-    
-  },[store?.theme]);
-  const user=useMemo(()=>store.user,[store.user]);
- 
+  const theme = useMemo(() => {
+    if (store?.theme === "dark")
+      return {
+        background: "#1b1d21",
+        color: "white",
+      };
+    return { background: "#f3f6f9" };
+  }, [store?.theme]);
+  const user = useMemo(() => store.user, [store.user]);
+
   return (
-    <div style={{height:'100vh' ,overflow:'scroll',...theme}}>
+    <div style={{ height: "100vh", overflow: "scroll", ...theme }}>
       <AppContext.Provider value={values}>
         <>
           <Loader loading={showLoader} />
           <MDBRow>
-           {user && <MDBCol size={2} className="p-0">
-              <SidebarComponent />
-            </MDBCol>}
-            <MDBCol size={user ? 10 : 12}>
+            {user && (
+              <MDBCol size={collapsed ? 1 : 2} className="p-0">
+                <SidebarComponent
```

Comment:
why are be handling collapsed from the App.js when we can handle this directly in the sidebar component

Reply:
When the sidebar collapses, we need to use the space which was occupied by the sidebar for the main page body. So, having it in App.js helps to do that.